services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviewsTable.js

```diff
@@ -0,0 +1,143 @@
+ */
+
+const propTypes = {
+  // eslint-disable-next-line
+  noticeOfWorkReviews: PropTypes.arrayOf(CustomPropTypes.NOWApplicationReview).isRequired,
+  noticeOfWorkReviewTypesHash: PropTypes.objectOf(PropTypes.string).isRequired,
+
+  handleDocumentDelete: PropTypes.func.isRequired,
+  handleDelete: PropTypes.func.isRequired,
+  openEditModal: PropTypes.func.isRequired,
+  handleEdit: PropTypes.func.isRequired,
+};
+
+const columns = [
+  {
+    title: "Type",
+    dataIndex: "now_application_review_type",
+    key: "now_application_review_type",
+  },
+  {
+    title: "Referee Name",
+    dataIndex: "referee_name",
+    key: "referee_name",
+  },
+  {
+    title: "Response Recieved",
+    dataIndex: "response_date",
+    key: "response_date",
+  },
+  {
+    title: "Documents",
+    dataIndex: "documents",
+    key: "documents",
+    render: (text) => (
+      <div title="Documents">
+        <ul>
+          {text.length > 0 &&
+            text.map((doc) => (
+              <li key={doc.mine_document.mine_document_guid}>
+                <div>
+                  <LinkButton
+                    key={doc.mine_document.mine_document_guid}
+                    onClick={() => downloadFileFromDocumentManager(doc.mine_document)}
+                  >
+                    {doc.mine_document.document_name}
+                  </LinkButton>
+                </div>
+              </li>
+            ))}
+        </ul>
+      </div>
+    ),
+  },
+  {
+    title: "",
+    dataIndex: "editDeleteButtons",
+    key: "editDeleteButtons",
+    align: "right",
+    render: (text, record) => (
+      <AuthorizationWrapper permission={Permission.EDIT_PERMITS}>
+        <Button
+          ghost
+          type="primary"
+          size="small"
+          onClick={(event) =>
+            record.openEditModal(event, record, record.handleEdit, record.handleDocumentDelete)
+          }
+        >
+          <img src={EDIT_OUTLINE_VIOLET} alt="Edit Review" />
+        </Button>
+        <Popconfirm
+          placement="topLeft"
+          title="Are you sure you want to delete this?"
+          onConfirm={() => record.handleDelete(record.now_application_review_id)}
+          okText="Delete"
+          cancelText="Cancel"
+        >
+          <Button ghost type="primary" size="small">
+            <Icon type="minus-circle" theme="outlined" />
+          </Button>
+        </Popconfirm>
+      </AuthorizationWrapper>
+    ),
+  },
+];
+
+const transformRowData = (
+  reviews,
+  reviewTypeHash,
+  handleDelete,
+  openEditModal,
+  handleEdit,
+  handleDocumentDelete
+) => {
+  return reviews.map((review) => ({
+    now_application_review_type: reviewTypeHash[review.now_application_review_type_code],
+    handleDelete,
+    openEditModal,
+    handleEdit,
+    handleDocumentDelete,
+    ...review,
+  }));
+};
+
+export const NOWApplicationReviewsTable = (props) => {
+  return (
+    <div>
+      <Table
```

Comment:
This table should be wrapped in a <TableLoadingWrapper> with an `isLoaded` condition to match the loading states of every table.

Reply:
added skeleton loading